### ISSUE TITLE
MongoParseError: Text record must only set `authSource` or `replicaSet`

### ISSUE THREAD

**Issue by jugaldb** [Automattic/mongoose#10572]
I am currently using version `5.13.7` of mongoose

The bug is that mongodb released their official driver to support serverless atlas instaces a few days ago in `version 4.1.0` of the official release. 

It connects fine using the mongodb driver but when I try to connect it using mongoose it throws me an error as follows:

### **MongoParseError: Text record must only set `authSource` or `replicaSet`**

My code for connecting is as follows:

```
const mongoose = require("mongoose");
const MONGO_CONSTANTS = require("../constants/database");
require("dotenv").config();

const URL = proccess.env.MONGO_URL;

(async () => {
  try {
    await mongoose.connect(URL, {
      useNewUrlParser: true,
      useFindAndModify: false,
      useCreateIndex: true,
      useUnifiedTopology: true,
    });
  } catch (e) {
    console.log("error occured: ", e.toString());
  }
})();

const connection = mongoose.connection;

connection.once(MONGO_CONSTANTS.OPEN_EVENT, () => {
  console.log(`Successfully connected to database at ${URL}`);
});

connection.on(MONGO_CONSTANTS.DISCONNECTED_EVENT, () => {
  console.log(`disconnected event to database at ${URL}`);
});

connection.on(MONGO_CONSTANTS.RECONNECT_FAILED_EVENT, () => {
  console.log(`reconnectFailed event to database at ${URL}`);
});

connection.on(MONGO_CONSTANTS.ERROR_EVENT, () => {
  console.log(`database connection error while connecting at ${URL}`);
});
```

The underlying issue is that mongoose isnt yet using the latest driver of mongodb for nodejs, Please fix this issue asap if possible. and upgrade to latest drivers

See: [Mongodb Forum link](https://www.mongodb.com/community/forums/t/atlas-serverless-and-dns-txt-record/117967) for more reference

**Comment by jugaldb** [comment]
I couldn't find any such issue, can you maybe provide a link for the same?

**Comment by ulf881** [comment]
> I couldn't find any such issue, can you maybe provide a link for the same?
> @IslandRhythms

#10560

Fix released on version 6.0.0-rc1

**Comment by jugaldb** [comment]
Thanks for the update, closing the issue
